### PR TITLE
 Use secrets for credentials for GCP

### PR DESCRIPTION
**Pull request by nikhita**
Follow up of https://github.com/kubermatic/kubermatic/pull/3836
/hold

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhita** [comment]
Needs https://github.com/kubermatic/kubermatic/pull/3964 first

**Comment by nikhita** [comment]
Ready for review.

File: api/pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -447,6 +447,8 @@ type PacketCloudSpec struct {
 
 // GCPCloudSpec specifies access data to GCP.
 type GCPCloudSpec struct {
+	CredentialsReference *providerconfig.GlobalSecretKeySelector `json:"credentialsReference,omitempty"`
+
 	ServiceAccount string `json:"serviceAccount"`
```
Comment (alvaroaleman) [inline api/pkg/crd/kubermatic/v1/cluster.go]: Can we mark the `serviceAccount` `omitempty`?

Comment (nikhita) [reply api/pkg/crd/kubermatic/v1/cluster.go]: done

**Comment by nikhita** [comment]
/assign @zreigz 
for approval

**Comment by zreigz** [comment]
I've tested it locally and it seems to doesn't work. The only etcd is created. In Dashboard I have error:
`failed unmarshalling service account: unexpected end of JSON input` 
Did you test it on your side?

**Comment by zreigz** [comment]
I've tested it on my local environment together with controllers. Works fine. 
Thanks for this PR :+1:

**Pull request merged**